pkg/initializer/initializer.go

```diff
@@ -46,8 +46,12 @@ func (e *EtcdInitializer) Initialize() error {
 		return err
 	}
 	if dataDirStatus != validator.DataDirectoryValid {
-		if err := e.restoreCorruptData(); err != nil {
-			return fmt.Errorf("error while restoring corrupt data: %v", err)
+		if rerr := e.restoreCorruptData(); rerr != nil {
+			dataDir := e.Config.RestoreOptions.RestoreDataDir
+			if err := removeContents(filepath.Join(dataDir)); err != nil {
```

Comment:
Why not restore to a different directory and rename/move the directory once restore is complete?
This is a widely used practice and for good reasons. For example, it doesn't create open questions about what happens if the delete fails, etc.

Reply:
Ah. Yes. Thats better.